rfcs/0000-top-level-assumptions.md

```diff
@@ -298,9 +298,9 @@ foo?.bar;
 
 # Open questions
 
-- Should we try to pass to the presets at least a _partial_ `assumptions` object? Currently `@babel/preset-env` relies on `loose` to enable/disable the `typeof-symbol` plugin.
-- Should we validate the list of `assumptions` in `@babel/core`, and disallow unknown ones? This would make it impossible for third-party plugins to introduce their own assumptions, but it also means that it's easier for us to introduce new assumptions without risking ecosystem incompatibilities.
-- Should assumptions always default to `false`? Currently everything defaults to being spec-compliant, except for the `arrow-functions` plugin which has a `spec: true` option as an opt-in.
+- Should we try to pass to the presets at least a _partial_ `assumptions` object? Currently `@babel/preset-env` relies on `loose` to enable/disable the `typeof-symbol` plugin. **ANSWER:** No, presets can only _produce_ assumptions and not _consume_ them.
```

Comment:
Why not? Must a preset provide an unconditional transform that does consume the assumptions, if it wishes to operate on them?

what if a preset is invalid when certain assumptions are made or not made?

Reply:
We had to choose: we either resolve assumptions _after_ resolving presets, or _before_ resolving presets.
In the first cast it's impossible to pass the resolved assumptions to presets (because we don't know them yet), in the second case it's impossible to set assumptions in presets (because they have already been resolved).

If a preset needs to change its behavior based on an assumption, I suggest moving that change down to the plugins that the preset enables.

Reply:
meaning, the plugin would validate and throw an error at transform time, not at config parsing time?

Reply:
At plugin instantiation time:

```js
function myPlugin(api) {
  if (api.assumption("unsupportedAssumption")) {
    throw new Error();
  }

  return {
    visitor: { ... }
  };
}
```

this is in the same phase as "preset instantiation" (it happens after, but you cannot only trigger the preset instantiation phase but not the plugin instantiation phase using the Babel API).